### PR TITLE
ledger doc store fix for archived products / locations

### PR DESCRIPTION
**Pull request by snopoke**
Use unfiltered Django manager in doc store to prevent doc not found errors.
@czue 

Outstanding question: Should stock state for archived products and locations be removed from ES?

### CODE REVIEW

File: corehq/form_processor/document_stores.py [hunk 1]
```diff
@@ -106,4 +106,4 @@ def __init__(self, domain):
         def _doc_gen_fn(obj):
             return obj.to_json()
 
-        super(LedgerV1DocumentStore, self).__init__(StockState, _doc_gen_fn)
+        super(LedgerV1DocumentStore, self).__init__(StockState, _doc_gen_fn, model_manager=StockState.include_archived)
```
Comment (dimagimon) [inline corehq/form_processor/document_stores.py]: E501 line too long (119 > 115 characters)

File: corehq/form_processor/tests/test_ledgers.py [hunk 2]
```diff
@@ -211,3 +213,67 @@ def _assert_transactions(self, values, ignore_ordering=False):
...
+        self.addCleanup(toggle_product_archive)
+
+        self.assertTrue(SQLProduct.objects.get(code=self.product_a.code).is_archived)
+
+        state = stock_states[0]
+        store = LedgerV1DocumentStore(DOMAIN)
+        doc = store.get_document(state.id)
+        self.assertEquals(int(doc['_id']), state.id)
+        self.assertEquals(doc['case_id'], state.case_id)
+
```
Comment (dimagimon) [inline corehq/form_processor/tests/test_ledgers.py]: W391 blank line at end of file

**Review by czue — APPROVED** [review]

**Comment by czue** [comment]
> Should stock state for archived products and locations be removed from ES?

I don't have an obvious answer but my guess is maybe not removed but tagged as archived? I think maybe we can just do nothing until it comes up in a reports context though? @esoergel might also have an opinion

**Comment by esoergel** [comment]
Tagged as archived (and filtered out in the default `LedgerES`) seems like a safe bet.

I also can't think of a good reason to keep them in elasticsearch, since presumably the only reason we'd query against them would be to restore them, which should be easy enough also in SQL.

**Pull request merged**